sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -207,16 +207,21 @@ public void testInWithSubqueryOrderImplicitlyAppliedOrdersMerged() {
         LogicalPlan plan = plan("select x from t1 where x in (select x from t1 order by x desc)");
         assertThat(plan.dependencies().entrySet().size(), is(1));
         LogicalPlan subPlan = plan.dependencies().keySet().iterator().next();
-        assertThat(subPlan, isPlan("RootBoundary[x]\nOrderBy[x][false][false]\nCollect[doc.t1 | [x] | All]\n"));
+        assertThat(subPlan, isPlan("RootBoundary[x]\n" +
+                                   "OrderBy[x][false][false]\n" +
+                                   "Collect[doc.t1 | [x] | All]\n"));
     }
 
     @Test
     public void testInWithSubqueryOrderWithExistingOrderByImplicitlyApplied() {
         LogicalPlan plan = plan("select x from t1 where x in (select x from t1 order by 1 desc limit 10)");
         assertThat(plan.dependencies().entrySet().size(), is(1));
         LogicalPlan subPlan = plan.dependencies().keySet().iterator().next();
-        assertThat(subPlan, isPlan("RootBoundary[x]\nOrderBy[x][false][false]\nLimit[10;0]\n" +
-                                   "OrderBy[x][true][null]\nCollect[doc.t1 | [x] | All]\n"));
+        assertThat(subPlan, isPlan("RootBoundary[x]\n" +
+                                   "OrderBy[x][false][false]\n" +
```

Comment:
Would it make sense to first check if the result is ordered before adding an additional order by?

Reply:
If it was the same ordering it should be merged,
here the query has a `DESC` order by.

Reply:
The more problematic would be if the subselect has the same OrderBy and a Limit:
`select x from t1 where x in (select x from t1 order by x limit 10)`
then we will have: Order[Limit[Order[...]]]
but this is an optimization we need to do generally for the tryOptimize, that when the OrderBy is exactly the same we can push it down after Limit and actually eliminate it.

Reply:
My point here is that we don't need to have a specific kind of ordering. Any order is fine(?). So if the subrelation already has an order we can just plan it without adding an additional order operator at all.
So it might make sense to have an additional `selectSymbol.relation().orderBy() != null` - this will also only work for simple cases where the subquery is not nested itself. But I think it would avoid ordering twice in more common cases.

Reply:
TermInSetQuery does a natural sorting, so it means that we need to check the orderBy() that has as first symbol the output symbol of the query and that it's sorted ASC. Does it make sense to add that "complex" check?

Reply:
Adding the check might be worthwhile. I would otherwise think about skipping the optimization if we make queries worse.

Reply:
The issue is that we make maybe the query slower but this query is executed once.
The TermInSetQuery is built many times (as many as the shards accessed from the outer query) so it still makes sense to do that.